consul/src/main/java/com/linecorp/armeria/internal/consul/HealthClient.java

```diff
@@ -0,0 +1,171 @@
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.consul;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.consul.ConsulClient;
+
+/**
+ * Health client supports below APIs.
+ * {@code HealthClient} is responsible for endpoint of Consul API:
+ * {@code `/health`}(https://www.consul.io/api/health.html)
+ *
+ * <p>GET /health/node/:node
+ * GET /health/checks/:service
+ * GET /health/service/:service?passing
+ * GET /health/connect/:service
+ * GET /health/state/:state
+ */
+public class HealthClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(HealthClient.class);
+
+    private final ConsulClient client;
+
+    public static HealthClient of(ConsulClient consulClient) {
+        return new HealthClient(consulClient);
+    }
+
+    protected HealthClient(ConsulClient client) {
+        this.client = client;
+    }
+
+    /**
+     * Gets a list of nodes for service.
+     */
+    private CompletableFuture<List<HealthService>> service(String serviceName, ConsulParameters parameters) {
+        logger.trace("service() {}, {}", serviceName, parameters.toParameterString());
+        return client.get("/health/service/" + serviceName + '?' + parameters.toParameterString())
+                     .aggregate()
+                     .thenApply(response -> {
+                         final List<HealthService> healths = new ArrayList<>();
+                         logger.trace("response: {}", response);
+                         if (!Objects.equals(response.status(), HttpStatus.OK)) {
+                             return healths;
+                         }
+                         final ObjectMapper mapper = new ObjectMapper();
+                         final HealthService[] resultHealthServices;
+                         try {
+                             resultHealthServices =
+                                     mapper.readValue(response.content().toStringUtf8(), HealthService[].class);
+                         } catch (IOException e) {
+                             return healths;
+                         }
```

Comment:
Could simplify with:
```java
try {
    return ImmutableList.copyOf(mapper.readValue(response.content().toStringUtf8(),
                                                 HealthService[].class));
} catch (IOException e) {
   return ImmutableList.of();
}
```

What do you think? 😀

Reply:
Wow, that's cool!
I like it.